src/CalcViewModel/DateCalculatorViewModel.cpp

```diff
@@ -320,43 +332,56 @@ String^ DateCalculatorViewModel::GetDateDiffString() const
     {
         if (addDelimiter)
         {
-            result = String::Concat(result, m_listSeparator);
+            result += m_listSeparator;
         }
         else
         {
             addDelimiter = true;
         }
 
-        result = String::Concat(result, String::Concat(GetLocalizedNumberString(dayCount), L" "));
+        result += GetLocalizedNumberString(dayCount)->Data();
+        result += L" ";
 
         if (dayCount > 1)
         {
-            result = String::Concat(result, resourceLoader.GetResourceString(L"Date_Days"));
+            result += resourceLoader.GetResourceString(L"Date_Days")->Data();
         }
         else
         {
-            result = String::Concat(result, resourceLoader.GetResourceString(L"Date_Day"));
+            result += resourceLoader.GetResourceString(L"Date_Day")->Data();
         }
     }
 
-    return result;
+    return ref new Platform::String(result.data());
 }
 
 String^ DateCalculatorViewModel::GetDateDiffStringInDays() const
 {
-    String^ strDateUnit;
+    wstring result;
```

Comment:
Why not use a `wstringstream`?

Reply:
For this kind of operation (limited number of small wchar_t[] to concatenate), std::(w)string::append is a lot faster than std::(w)stringstream.

See here: https://stackoverflow.com/questions/18892281/most-optimized-way-of-concatenation-in-strings

Reply:
Yes, I should read up on these things before commenting :)  I have been using `stringstream` because I thought it was standard.  TIL!